src/resources/js/accordion.js

```diff
@@ -0,0 +1,52 @@
+(function( window, $ ) {
+	var accordions = document.getElementsByClassName( 'accordion' );
+
+	var accordionIndex = ( undefined === accordionIndex ) ? 0 : accordionIndex++;
```

Comment:
An excellent pile 'o questions. As much as I like snake_case, the JS standard is camelCase and (as you point out, @barryhughes) the WP standard is now camelCase. Perhaps it is time to move to camelCase in our JS? Let's chat about it next Wednesday on our call.